graylog2-web-interface/src/components/common/ScrollContainer/useScrollContainer.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { useContext } from 'react';
+
+import ScrollContainerContext from './ScrollContainerContext';
+
+const useScrollContainer = () => {
+  const contextValue = useContext(ScrollContainerContext);
+
+  if (!contextValue) {
+    throw new Error('useScrollContainer hook needs to be used inside ScrollContainerContext.Provider');
+  }
+
+  return contextValue;
+};
+
+export default useScrollContainer;
```

Comment:
Instead of having this hook, the related context and provider we could also maintain the container ref in the `Search` component and pass it to the `SearchBar` and `DashboardSearchBar`, which provides it for the `ScrollToHint` component.

But I wanted to avoid prop drilling and allow reusing the `ScrollToHint` component easily on other pages.